ui/hooks/confirm/syncConfirmPath.test.ts

```diff
@@ -0,0 +1,33 @@
+import { renderHookWithProvider } from '../../../test/lib/render-helpers';
+import syncConfirmPath from './syncConfirmPath';
+
+jest.mock('react-router-dom', () => ({
+  ...jest.requireActual('react-router-dom'),
+  useHistory: () => ({ replace: jest.fn() }),
+}));
+
+const mockState = {
+  confirm: {
+    currentConfirmation: {
+      id: '1',
+      msgParams: {},
+    },
+  },
+};
+
+describe('syncConfirmPath', () => {
+  it('should execute correctly', () => {
+    const result = renderHookWithProvider(() => syncConfirmPath(), mockState);
+    expect(result).toBeDefined();
+  });
+
+  // for some weird reason this test is failing
```

Comment:
The issue with this is because the inner mock in this test case is not being called, but the outer mock on line 6. To fix this

1. I created a mock variable on line 4 `const mockHistoryReplace = jest.fn();`
2. I set the replace on line 8 to mock variable `useHistory: () => ({ replace: mockHistoryReplace }),`
3. I rewrote the test below as 
```
  it('should replace history route', async () => {
    mockHistoryReplace.mockClear();
    renderHookWithProvider(() => syncConfirmPath(), mockState);
    expect(mockHistoryReplace).toHaveBeenCalled();
  });
  ````

Reply:
Thanks I made this update.